lib/bundler/definition.rb

```diff
@@ -904,7 +904,7 @@ def expand_dependencies(dependencies, remote = false)
             "To add those platforms to the bundle, " \
             "run `bundle lock --add-platform #{mapped_platforms.join " "}`."
         end
-        platforms.each do |p|
+        Resolver.sort_platforms(platforms).each do |p|
```

Comment:
Should be improved now, it only sorts the platforms once instead of once per dependency